File: Data Structures & Algorithms/two-integer-sum-ii/submission-2.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int left = 0;
        int right = numbers.length-1;
        for (int i =0;i<numbers.length;i++){
            if (numbers[left]+numbers[right]==target){
                return new int [] {1+left,1+right};
            }
            if (numbers[left]+numbers[right]>target){
                right--;
            }
            if (numbers[left]+numbers[right]<target){
                left++;
            }
        }
        return new int [] {numbers[left],numbers[right]};
    }
}

File: Data Structures & Algorithms/two-integer-sum/submission-2.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        check = {}
        for i, num in enumerate(nums):
            if (check.get(target-num) == None):
                check[num] = i
            else:
                return [check.get(target-num),i]
        return [0,0]